<launch>
  <arg name="REMOTE_TF" default="/tf_low_frequency"/>
  <arg name="USE_LHAND_POINTCLOUD" default="true" />
  <arg name="USE_RHAND_POINTCLOUD" default="true" />
  <arg name="ROI_MARKER_BASEFRAME" default="pelvis" />

  <group ns="pointcloud_roi">
    <remap from="/tf" to="$(arg REMOTE_TF)"/>
    <node pkg="jsk_interactive_marker"
          type="point_cloud_config_marker"
          name="roi_point_cloud_config_marker"
          output="log"
          clear_params="true">
      <param name="base_frame" value="$(arg ROI_MARKER_BASEFRAME)" />
      <param name="size" value="0.5" />
      <remap from="~get" to="add_grid" />
    </node>


    <node pkg="nodelet" type="nodelet"
          name="pcl_roi_remote_manager" args="manager"
          output="log" alaunch-prefix="xterm -e gdb --args" />


    <node pkg="nodelet" type="nodelet"
          name="voxel_grid_decoder"
          args="load jsk_pcl/VoxelGridDownsampleDecoder /pointcloud_roi/pcl_roi_remote_manager"
          output="screen" clear_params="true">
      <remap from="~input" to="voxel_grid_manager/output_encoded" />
    </node>


    <!-- left hand -->


    <node if="$(arg USE_LHAND_POINTCLOUD)"
	  pkg="nodelet" type="nodelet"
          name="l_hand_voxel_grid_decoder"
          args="load jsk_pcl/VoxelGridDownsampleDecoder /pointcloud_roi/pcl_roi_remote_manager"
          output="screen" clear_params="true">
      <remap from="~input" to="l_hand_voxel_grid_manager/output_encoded" />
    </node>


    <!-- right hand -->

    <node if="$(arg USE_RHAND_POINTCLOUD)"
	  pkg="nodelet" type="nodelet"
          name="r_hand_voxel_grid_decoder"
          args="load jsk_pcl/VoxelGridDownsampleDecoder /pointcloud_roi/pcl_roi_remote_manager"
          output="screen" clear_params="true">
      <remap from="~input" to="r_hand_voxel_grid_manager/output_encoded" />
    </node>

  </group>
</launch>
